siddhi-function-archetype/src/main/resources/archetype-resources/src/main/java/__siddhi_function_name__Function.java

```diff
@@ -0,0 +1,154 @@
+                    "for the argument of ${siddhi_function_name} function, " +
+                    "required " + Attribute.Type.INT + " or " + Attribute.Type.LONG +
+                    " or " + Attribute.Type.FLOAT + " or " + Attribute.Type.DOUBLE +
+                    ", but found " + attributeType.toString());
+        }
+    }
+
+    /**
+     * The main execution method which will be called upon event arrival
+     * when there are more than one function parameter
+     *
+     * @param data the runtime values of function parameters
+     * @return the function result
+     */
+    @Override
+    protected Object execute(Object[] data) {
+        return null;
+    }
+
+    /**
+     * The main execution method which will be called upon event arrival
+     * when there are zero or one function parameter
+     *
+     * @param data null if the function parameter count is zero or
+     *             runtime data value of the function parameter
+     * @return the function result
+     */
+    @Override
+    protected Object execute(Object data) {
+        //Sample code
+        if (data != null) {
+            //type-conversion
+            if (data instanceof Integer) {
+                int inputInt = (Integer) data;
+                return ((double) inputInt / KILOMETERS_PER_MILE);
+            } else if (data instanceof Long) {
+                long inputLong = (Long) data;
+                return ((double) inputLong / KILOMETERS_PER_MILE);
+            } else if (data instanceof Float) {
+                float inputFloat = (Float) data;
+                return ((double) inputFloat / KILOMETERS_PER_MILE);
+            } else if (data instanceof Double) {
+                return ((Double) data / KILOMETERS_PER_MILE);
+            }
+        } else {
+            throw new ExecutionPlanRuntimeException("Input to the ${siddhi_function_name} function cannot be null");
+        }
+        return null;
+    }
+
+    /**
+     * This will be called only once and this can be used to acquire
+     * required resources for the processing element.
+     * This will be called after initializing the system and before
+     * starting to process the events.
+     */
+    @Override
+    public void start() {
+        //Implement start logic to acquire relevant resources
+    }
+
+    /**
+     * This will be called only once and this can be used to release
+     * the acquired resources for processing.
+     * This will be called before shutting down the system.
+     */
+    @Override
+    public void stop() {
+        //Implement stop logic to release the acquired resources
+    }
+
+    @Override
+    public Attribute.Type getReturnType() {
+        return Attribute.Type.DOUBLE;
+    }
+
+
+    /**
+     * Used to collect the serializable state of the processing element, that need to be
+     * persisted for the reconstructing the element to the same state on a different point of time
```

Comment:
Require full stop in ending main comment line(class level, methods etc..). Fix in all places.